lib/execution_engine2/authorization/authstrategy.py

```diff
@@ -0,0 +1,139 @@
+    :param job: a Job model object
+    :param user_id: string - the user id
+    :param token: string - the user's auth token
+    :param config: dict - the service config
+    :returns: bool - True if the user can read the job info
+    """
+    return _check_permissions(job, user_id, token, config, level="read")
+
+
+def can_write_job(job: Job, user_id: str, token: str, config: Dict[str, str]) -> bool:
+    """
+    Returns True if the user has write access to the job, False otherwise.
+    :param job: a Job model object
+    :param user_id: string - the user id
+    :param token: string - the user's auth token
+    :param config: dict - the service config
+    :returns: bool - True if the user can read the job info
+    """
+    return _check_permissions(job, user_id, token, config, level="write")
+
+
+def can_read_jobs(jobs: List[Job], user_id: str, token: str, config: Dict[str, str]) -> List[bool]:
+    """
+    Returns a list of job permissions in the same order as the given list of Jobs.
+    :param job: a Job model object
+    :param user_id: string - the user id
+    :param token: string - the user's auth token
+    :param config: dict - the service config
+    :returns: List[bool] - Has True values if the user can read job info, False otherwise
+    """
+    return _check_permissions_list(jobs, user_id, token, config, level="read")
+
+
+def can_write_jobs(jobs: List[Job], user_id: str, token: str, config: Dict[str, str]) -> List[bool]:
+    """
+    Returns a list of job write permissions in the same order as the given list of Jobs.
+    :param job: a Job model object
+    :param user_id: string - the user id
+    :param token: string - the user's auth token
+    :param config: dict - the service config
+    :returns: List[bool] - Has True values if the user can write job info, False otherwise
+    """
+    return _check_permissions_list(jobs, user_id, token, config, level="write")
+
+
+def _check_permissions(job: Job, user_id: str, token: str, config: Dict[str, str], level="read") -> bool:
+    """
+    Returns a job permissions, for either read or write ability
+    :param job: a Job model object
+    :param user_id: string - the user id
+    :param token: string - the user's auth token
+    :param config: dict - the service config
+    :param level: string - if "read", then returns the read value, if "write", return whether the user can write.
+    :returns: bool - True if the permission is valid, False otherwise.
+    """
+    if user_id == job.user:
+        return True
+    if job.authstrat == KBASE_WS_AUTHSTRAT:
+        ws_auth = WorkspaceAuth(token, user_id, config['workspace-url'])
+        if level == "read":
+            return ws_auth.can_read(job.wsid)
+        else:
+            return ws_auth.can_write(job.wsid)
+    else:
+        return user_id == job.user
+
+
+def _check_permissions_list(jobs: List[Job], user_id: str, token: str, config: Dict[str, str], level="read") -> bool:
```

Comment:
I think the return type is not a bool, but a list of bool

Reply:
Yes. Thanks for catching that.